python/imageutils.py

```diff
@@ -295,125 +185,55 @@ def reconstruct_block_slice(self, reconstructed, legend, mem):
-                            data = np.concatenate((data,block.get_data()), 2)
-                        else:
-                            raise Exception('Unable to concatenate data')
-                    except Exception as e:
-                        print 'ERROR: ', e
-                        sys.exit(1)
-                        
-                remaining_mem = remaining_mem - (block_bytes)
+                total_read += read_time
 
                 #cannot load another block into memory, must write slice
-                if remaining_mem / block_bytes < 1:
+                if remaining_mem <= 0:
                     print "Remaining memory:{0}".format(remaining_mem)
 
-                    end_x = int(block_pos[-1]) + block_shape[2]
-                    end_z = int(block_pos[-2]) + block_shape[1]
-                    end_y = int(block_pos[-3]) + block_shape[0]
-
-                    print "Writing data starting at ({0}, {1}, {2}) and ending at ({3}, {4}, {5})".format(start_y, start_z, start_x, end_y, end_z, end_x)
-                    for i in range(0, block_shape[2]):
-                        seek_start = time()
-                        reconstructed.seek(self.header_size + bytes_per_voxel*(start_y + (start_z)*y_size +(start_x + i)*y_size*z_size), 0)
-                        print "Seek time for slice of  depth {0}: {1}".format(start_x + i, time()-seek_start)
-                        write_start = time()
-                        reconstructed.write(data[:,:,i].tobytes('F'))
-                        print "Write time for slice of  depth {0}: {1}".format(start_x + i, time()-write_start)
+                    seek_time, defrag_time, write_time = write_to_file(data_dict, reconstructed, bytes_per_voxel)
+                    
+                    total_seek += seek_time
+                    total_defrag += defrag_time
+                    total_write += write_time
                     
-
                     remaining_mem = mem
-                    data = None
-
-                prev_block_start = (block_pos[-3], block_pos[-2], block_pos[-1])
-
-
-    #TODO: Fix function so that it will work in HDFS
-    def reconstruct_slice_slice(self, reconstructed, legend, mem=None):
-        """ Reconstructs and image given a set of slices.
-
-        Keyword arguments:
-        reconstructed                   : the fileobject pointing to the to-be reconstructed image
-        legend                          : a legend containing the location of the blocks to use for reconstruction.
-        mem                             : the amount of available memory in bytes in the system. If None is specified, only one slice will be loaded into memory at a time
 
-        NOTE: currently only supports nifti slices as it uses 'bitpix' to determine number of bytes per voxel. Element is specific to nifti headers
+            #if there is a final chunk of data that remains to be copied
```

Comment:
I had checked yesterday what was the best way to determine if i've reached eof in python (which would enable me to mutualize the code), and i must have been really tired because I completely misunderstood what I was reading. Will fix this!